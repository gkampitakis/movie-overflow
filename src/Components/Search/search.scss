.search_container {
  height: 15%;
  position: relative;
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
    url('../../assets/homeImg.jpg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;

  transition: height 0.7s linear;

  &.home {
    /* Position and center the image to scale nicely on all screens */
    height: 50%;

    h1 {
      top: 60%;
    }
  }

  h1 {
    font-family: 'Haas Grot Text R Web', 'Helvetica Neue', Helvetica, Arial,
      sans-serif;
    font-size: 48px;
    font-weight: 700;
    line-height: 56px;
    color: white;
    text-align: center;
    top: 3vh;
    position: inherit;
    margin-bottom: 50px;

    transition: top 0.7s linear;
  }

  .search {
    .search_input {
      text-align: left;
      padding-left: 50px;
      padding-right: 50px;
      width: 30%;
      height: 64px;
      font-size: 16px;
      box-shadow: 0px 8px 20px rgba(0, 0, 0, 0.06);
      border-radius: 8px;
      border: none;

      &:focus,
      :active,
      :focus-visible {
        outline: none;
      }
    }

    .search_icon {
      cursor: pointer;
      position: relative;
      left: 40px;
      font-size: 25px;
      transform: translate(0%, 25%);
      color: rgba(60, 60, 60, 0.85);
    }

    .close_icon {
      visibility: hidden;
      cursor: pointer;
      position: relative;
      right: 40px;
      font-size: 25px;
      transform: translate(0%, 25%);
      color: rgba(60, 60, 60, 0.85);

      &.visible {
        visibility: unset;
      }
    }

    width: 100%;
    text-align: center;
    position: absolute;
    top: 100%;
    left: 50%;
    margin: auto;
    transform: translate(-50%, -50%);
  }

  .filters {
    justify-content: space-evenly;
    display: flex;
    width: 30%;
    margin: auto;
    margin-top: 10px;
    label {
      font-size: 16px;
      padding-left: 5px;
    }

    input[type='checkbox'] {
      width: 15px;
      height: 15px;
      border-radius: 4px;
      border: 0.1mm solid #929292;
    }
  }
}
