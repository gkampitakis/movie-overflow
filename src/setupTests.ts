// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import fetchMock from 'jest-fetch-mock';

fetchMock.enableMocks();

export const movieDetailsResponse = {
  id: 'mockId',
  original_title: 'mockTitle',
  runtime: 100,
  genres: [{ name: 'test' }],
  poster_path: '/mock_path',
  tagline: 'mockTagLine',
  overview: 'mockOverview',
  status: 'mockStatus',
  release_date: 'mockDate',
  vote_average: 100,
  images: {
    posters: [
      {
        file_path: '/mock_image_path'
      }
    ]
  },
  credits: {
    cast: [
      {
        id: 'id1',
        profile_path: 'cast_path',
        original_name: 'mockName',
        character: 'character1'
      },
      {
        id: 'id2',
        profile_path: null,
        original_name: 'mockName2',
        character: 'character2'
      }
    ],
    crew: [
      {
        id: 'crewId1',
        profile_path: 'crew_path',
        original_name: 'crewMockName',
        job: 'job1'
      },
      {
        id: 'crewId2',
        profile_path: null,
        original_name: 'crewMockName2',
        job: 'job2'
      }
    ]
  }
};