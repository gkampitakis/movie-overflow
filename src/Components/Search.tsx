import React, { useState } from 'react';
import { BsSearch } from 'react-icons/bs';
import { IoMdClose } from 'react-icons/io';
import './search.scss';

export default function Search() {
  const [isHome, setIsHome] = useState(true);
  const [value, setValue] = useState('');
  const [filters, setFilters] = useState({
    actors: true,
    movies: true,
    series: true
  });
  const clearInput = () => {
    setValue('');
    setIsHome(true);
  };

  const handleFiltersChange = (field: 'actors' | 'movies' | 'series') => {
    setFilters((filters) => ({ ...filters, [field]: !filters[field] }));
  };

  const search = () => {
    if (!value) return;
    setIsHome(false);
    console.log('close');
  };

  const enterHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
    if (e.key === 'Enter') {
      search();
    }
  };

  return (
    <section className={`search_container ${isHome && 'home'}`}>
      <h1>Movie Overflow</h1>
      <div className="search">
        <BsSearch type="submit" className="search_icon" onClick={search} />
        <input
          className="search_input"
          type="text"
          value={value}
          onKeyDown={enterHandler}
          onChange={(e) => setValue(e.target.value)}
          placeholder="Search..."
        />
        <IoMdClose
          onClick={clearInput}
          className={`close_icon ${!!value && 'visible'}`}
        />

        <div className="filters">
          <div>
            <input
              type="checkbox"
              id="movie"
              name="Movie"
              defaultChecked={filters.movies}
              onChange={() => handleFiltersChange('movies')}
            />
            <label htmlFor="Movie">Movie</label>
          </div>
          <div>
            <input
              type="checkbox"
              id="tv_series"
              name="Tv Series"
              defaultChecked={filters.series}
              onChange={() => handleFiltersChange('series')}
            />
            <label htmlFor="tv_series">Tv Series</label>
          </div>
          <div>
            <input
              type="checkbox"
              id="actor"
              name="Actor"
              defaultChecked={filters.actors}
              onChange={() => handleFiltersChange('actors')}
            />
            <label htmlFor="Actor">Actor</label>
          </div>
        </div>
      </div>
    </section>
  );
}
