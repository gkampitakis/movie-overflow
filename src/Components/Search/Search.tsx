import React, { useState, useEffect, useRef } from 'react';
import { BsSearch } from 'react-icons/bs';
import { IoMdClose } from 'react-icons/io';
import { useHistory, useLocation } from 'react-router-dom';
import { SearchOptions, Suggestion } from '../../types';
import { searchRequest } from '../../Api';
import { Loader, RadioButton } from '../';
import './search.scss';

const DEBOUNCE_TIMER = 500;

function extractSuggestions(data: any[], option: SearchOptions): Suggestion[] {
  if (!data.length) return [];
  const random = Math.floor(Math.random() * data.length);

  return data.slice(random, random + 5).map((elem) => {
    const media_type = elem.media_type ? elem.media_type : option;
    let title = elem.original_title;
    if (media_type === 'tv') {
      title = elem.original_name;
    } else if (media_type === 'person') {
      title = elem.name;
    }

    return {
      id: elem.id,
      title,
      media_type
    };
  });
}

export default function Search() {
  const inputRef = useRef<HTMLInputElement>(null);
  const history = useHistory();
  const { pathname } = useLocation();
  const [isHome, setIsHome] = useState(true);
  const [query, setQuery] = useState('');
  const [radioOption, setRadioOption] = useState<SearchOptions>('all');
  const [suggestions, setSuggestions] = useState<Suggestion[] | undefined>();
  const [loading, setLoading] = useState<boolean>(false);
  const searchDebounce = useRef<ReturnType<typeof setTimeout> | undefined>(
    undefined
  );

  useEffect(() => {
    if (pathname === '/') {
      return setIsHome(true);
    }

    const [, path, queryString] = pathname.split('/');
    if (path === 'search') {
      setQuery(queryString);
    }
  }, [pathname]);

  useEffect(() => {
    inputRef.current?.focus();
    handleSearch();
  }, []);

  const clearInput = () => {
    setQuery('');
    setSuggestions([]);
  };

  const handleRequest = () => {
    setLoading(true);

    searchRequest(query, 1, radioOption)
      .then((data) => {
        setSuggestions(extractSuggestions(data.results, radioOption));
      })
      .catch(console.error)
      .finally(() => setLoading(false));
  };

  const handleRadioChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setRadioOption(e.target.value as SearchOptions);
  };

  const handleSearch = () => {
    if (!query) return;
    setSuggestions([]);

    history.push({
      pathname: `/search/${query}`,
      state: { searchBy: radioOption }
    });
  };

  const navigate = (pathname: string) => {
    setSuggestions([]);

    history.push({
      pathname
    });
  };

  const keyboardHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
    if (e.key === 'Enter') {
      return handleSearch();
    }

    if (e.key === 'Escape' || e.key === 'Esc') {
      setSuggestions([]);
    }
  };

  const returnHome = () => {
    navigate('/');
    setQuery('');
    setRadioOption('all');
  };

  const handleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const input = e.target.value;
    setQuery(input);

    if (input.length <= 5) {
      setSuggestions([]);
      return;
    }

    if (searchDebounce.current) clearTimeout(searchDebounce.current);
    searchDebounce.current = setTimeout(() => {
      handleRequest();
    }, DEBOUNCE_TIMER);
  };

  return (
    <section className={`search_container ${isHome && 'home'}`}>
      <h1 onClick={returnHome}>Movie Overflow</h1>
      <div className="search">
        <div className="filters">
          <RadioButton
            name="All"
            value="all"
            checked={radioOption}
            onClick={handleRadioChange}
          />
          <RadioButton
            name="Movie"
            value="movie"
            checked={radioOption}
            onClick={handleRadioChange}
          />
          <RadioButton
            name="Tv Series"
            value="tv"
            checked={radioOption}
            onClick={handleRadioChange}
          />
          <RadioButton
            name="People"
            value="person"
            checked={radioOption}
            onClick={handleRadioChange}
          />
        </div>
        <div className="input_wrapper">
          <BsSearch
            type="submit"
            data-testid="search-icon"
            className="search_icon"
            onClick={handleSearch}
          />
          <input
            autoComplete="off"
            ref={inputRef}
            className="search_input"
            type="text"
            value={query}
            onKeyDown={keyboardHandler}
            onChange={handleInput}
            placeholder="Search..."
          />
          <IoMdClose
            data-testid="clear-icon"
            onClick={clearInput}
            className={`close_icon ${!!query && 'visible'}`}
          />
          {!!suggestions?.length && (
            <div className={`suggestions ${loading && 'loading'}`}>
              <Loader loading={loading} />
              {suggestions.map(({ media_type, id, title }) => (
                <div
                  onClick={() => navigate(`/${media_type}/${id}`)}
                  key={id}
                  className="item"
                >
                  <h3>{title}</h3>
                  <div className="type">{media_type}</div>
                </div>
              ))}
            </div>
          )}
        </div>
      </div>
    </section>
  );
}
