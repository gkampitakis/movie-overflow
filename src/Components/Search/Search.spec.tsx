import React from 'react';
import Search from '.';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { FetchMock } from 'jest-fetch-mock';
import { mockHistoryPush, mockResponseOnce } from '../../setupTests';

let mockPathname = '/';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  }),
  useLocation: () => ({
    pathname: mockPathname
  })
}));
const fetchMock = fetch as FetchMock;

describe('Search', () => {
  beforeEach(() => {
    mockPathname = '/';
    fetchMock.resetMocks();
    mockHistoryPush.mockClear();
    jest.spyOn(Math, 'random').mockImplementation(() => 0);
  });

  describe('When rendering', () => {
    it('Should render consistently', () => {
      const { container } = render(<Search />);

      expect(container).toMatchSnapshot();
    });

    it('Should have focused on search input', () => {
      render(<Search />);

      const searchInput = screen.getByPlaceholderText('Search...');
      expect(searchInput).toHaveFocus();
    });

    it('Should populate input with url query', () => {
      mockPathname = '/search/mockQuery';
      render(<Search />);

      const searchInput = screen.getByPlaceholderText('Search...');
      expect(searchInput).toHaveValue('mockQuery');
    });
  });

  it('Should return home', () => {
    render(<Search />);

    const title = screen.getByText(/movie overflow/i);
    const movieOption = screen.getByTestId('Movie-radio');
    const allOption = screen.getByTestId('All-radio');
    const searchInput = screen.getByPlaceholderText('Search...');

    userEvent.click(movieOption);
    userEvent.type(searchInput, 'hello world');
    expect(searchInput).toHaveValue('hello world');
    userEvent.click(title);

    expect(allOption).toBeChecked();
    expect(movieOption).not.toBeChecked();
    expect(searchInput).toHaveValue('');
    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: '/'
    });
  });

  it('Should clear input', () => {
    render(<Search />);
    const clearIcon = screen.getByTestId('clear-icon');
    const searchInput = screen.getByPlaceholderText('Search...');

    userEvent.type(searchInput, 'hello world');
    expect(searchInput).toHaveValue('hello world');

    userEvent.click(clearIcon);

    expect(searchInput).toHaveValue('');
  });

  it('Should trigger a search', () => {
    render(<Search />);

    const searchInput = screen.getByPlaceholderText('Search...');
    const searchIcon = screen.getByTestId('search-icon');

    userEvent.type(searchInput, 'my movie');
    userEvent.click(searchIcon);

    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: '/search/my movie',
      state: { searchBy: 'all' }
    });
  });

  it('Should trigger a search with Enter', () => {
    render(<Search />);

    const searchInput = screen.getByPlaceholderText('Search...');

    userEvent.type(searchInput, 'my movie');
    userEvent.type(searchInput, '{enter}');

    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: '/search/my movie',
      state: { searchBy: 'all' }
    });
  });

  describe('Search suggestions', () => {
    it('Should trigger search for suggestions', async () => {
      jest.useFakeTimers();
      mockResponseOnce({
        results: [
          {
            id: 1,
            media_type: 'person',
            name: 'mockActor'
          },
          {
            id: 2,
            media_type: 'tv',
            original_name: 'mockTvSeries'
          },
          {
            id: 3,
            original_title: 'mockMovie'
          }
        ]
      });
      const { container } = render(<Search />);
      const url =
        'https://api.themoviedb.org/3/search/multi?api_key=mockKey&query=my movi&page=1';
      const searchInput = screen.getByPlaceholderText('Search...');

      userEvent.type(searchInput, 'my movie');

      act(() => {
        jest.advanceTimersByTime(502);
      });

      await screen.findByText('mockTvSeries');

      expect(fetchMock).toHaveBeenCalledWith(url, {
        method: 'GET'
      });
      expect(container).toMatchSnapshot();
      jest.useRealTimers();
    });

    it('Should dismiss suggestions with ESC', async () => {
      jest.useFakeTimers();
      mockResponseOnce({
        results: [
          {
            id: 1,
            original_title: 'mockMovie'
          }
        ]
      });
      const { container } = render(<Search />);
      const url =
        'https://api.themoviedb.org/3/search/movie?api_key=mockKey&query=my movi&page=1';
      const searchInput = screen.getByPlaceholderText('Search...');
      const movieOption = screen.getByTestId('Movie-radio');

      userEvent.click(movieOption);
      userEvent.type(searchInput, 'my movie');

      act(() => {
        jest.advanceTimersByTime(502);
      });

      await screen.findByText('mockMovie');

      userEvent.type(searchInput, '{esc}');

      expect(fetchMock).toHaveBeenCalledWith(url, {
        method: 'GET'
      });
      expect(container).toMatchSnapshot();
      jest.useRealTimers();
    });

    it('Should dismiss suggestions when clearing input', async () => {
      jest.useFakeTimers();
      mockResponseOnce({
        results: [
          {
            id: 1,
            name: 'mockActor'
          }
        ]
      });
      const { container } = render(<Search />);
      const url =
        'https://api.themoviedb.org/3/search/person?api_key=mockKey&query=my perso&page=1';
      const searchInput = screen.getByPlaceholderText('Search...');
      const peopleOption = screen.getByTestId('People-radio');
      const clearIcon = screen.getByTestId('clear-icon');

      userEvent.click(peopleOption);

      userEvent.type(searchInput, 'my person');

      act(() => {
        jest.advanceTimersByTime(502);
      });

      await screen.findByText('mockActor');

      userEvent.click(clearIcon);

      expect(fetchMock).toHaveBeenCalledWith(url, {
        method: 'GET'
      });
      expect(container).toMatchSnapshot();
      jest.useRealTimers();
    });

    it('Should navigate to suggestion', async () => {
      jest.useFakeTimers();
      mockResponseOnce({
        results: [
          {
            id: 1,
            media_type: 'person',
            name: 'mockActor'
          },
          {
            id: 2,
            media_type: 'tv',
            original_name: 'mockTvSeries'
          },
          {
            id: 3,
            original_title: 'mockMovie',
            media_type: 'movie'
          }
        ]
      });
      const { container } = render(<Search />);
      const url =
        'https://api.themoviedb.org/3/search/multi?api_key=mockKey&query=my movi&page=1';
      const searchInput = screen.getByPlaceholderText('Search...');

      userEvent.type(searchInput, 'my movie');

      act(() => {
        jest.advanceTimersByTime(502);
      });

      const movieSuggestion = await screen.findByText('mockMovie');

      userEvent.click(movieSuggestion);

      expect(mockHistoryPush).toHaveBeenCalledWith({
        pathname: '/movie/3'
      });
      expect(fetchMock).toHaveBeenCalledWith(url, {
        method: 'GET'
      });
      expect(container).toMatchSnapshot();
      jest.useRealTimers();
    });

    it('Should show no suggestions when no results', async () => {
      jest.useFakeTimers();
      mockResponseOnce({
        results: []
      });

      const { container } = render(<Search />);
      const searchInput = screen.getByPlaceholderText('Search...');

      userEvent.type(searchInput, 'my movie');

      act(() => {
        jest.advanceTimersByTime(502);
        expect(container).toMatchSnapshot();
      });

      jest.useRealTimers();
    });
  });
});
