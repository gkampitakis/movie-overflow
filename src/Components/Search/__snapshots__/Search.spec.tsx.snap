// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search suggestions Should dismiss suggestions when clearing input 1`] = `
<div>
  <section
    class="search_container home"
  >
    <h1>
      Movie Overflow
    </h1>
    <div
      class="search"
    >
      <div
        class="filters"
      >
        <div
          class="radio_button"
        >
          <input
            checked=""
            data-testid="All-radio"
            id="all"
            name="All"
            type="radio"
            value="all"
          />
          <label
            for="all"
          >
            All
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Movie-radio"
            id="movie"
            name="Movie"
            type="radio"
            value="movie"
          />
          <label
            for="movie"
          >
            Movie
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Tv Series-radio"
            id="tv"
            name="Tv Series"
            type="radio"
            value="tv"
          />
          <label
            for="tv"
          >
            Tv Series
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="People-radio"
            id="person"
            name="People"
            type="radio"
            value="person"
          />
          <label
            for="person"
          >
            People
          </label>
        </div>
      </div>
      <div
        class="input_wrapper"
      >
        <svg
          class="search_icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          type="submit"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
            fill-rule="evenodd"
          />
        </svg>
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search..."
          type="text"
          value=""
        />
        <svg
          class="close_icon false"
          data-testid="clear-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`Search Search suggestions Should dismiss suggestions with ESC 1`] = `
<div>
  <section
    class="search_container home"
  >
    <h1>
      Movie Overflow
    </h1>
    <div
      class="search"
    >
      <div
        class="filters"
      >
        <div
          class="radio_button"
        >
          <input
            checked=""
            data-testid="All-radio"
            id="all"
            name="All"
            type="radio"
            value="all"
          />
          <label
            for="all"
          >
            All
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Movie-radio"
            id="movie"
            name="Movie"
            type="radio"
            value="movie"
          />
          <label
            for="movie"
          >
            Movie
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Tv Series-radio"
            id="tv"
            name="Tv Series"
            type="radio"
            value="tv"
          />
          <label
            for="tv"
          >
            Tv Series
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="People-radio"
            id="person"
            name="People"
            type="radio"
            value="person"
          />
          <label
            for="person"
          >
            People
          </label>
        </div>
      </div>
      <div
        class="input_wrapper"
      >
        <svg
          class="search_icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          type="submit"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
            fill-rule="evenodd"
          />
        </svg>
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search..."
          type="text"
          value="my movie"
        />
        <svg
          class="close_icon visible"
          data-testid="clear-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`Search Search suggestions Should navigate to suggestion 1`] = `
<div>
  <section
    class="search_container home"
  >
    <h1>
      Movie Overflow
    </h1>
    <div
      class="search"
    >
      <div
        class="filters"
      >
        <div
          class="radio_button"
        >
          <input
            checked=""
            data-testid="All-radio"
            id="all"
            name="All"
            type="radio"
            value="all"
          />
          <label
            for="all"
          >
            All
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Movie-radio"
            id="movie"
            name="Movie"
            type="radio"
            value="movie"
          />
          <label
            for="movie"
          >
            Movie
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Tv Series-radio"
            id="tv"
            name="Tv Series"
            type="radio"
            value="tv"
          />
          <label
            for="tv"
          >
            Tv Series
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="People-radio"
            id="person"
            name="People"
            type="radio"
            value="person"
          />
          <label
            for="person"
          >
            People
          </label>
        </div>
      </div>
      <div
        class="input_wrapper"
      >
        <svg
          class="search_icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          type="submit"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
            fill-rule="evenodd"
          />
        </svg>
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search..."
          type="text"
          value="my movie"
        />
        <svg
          class="close_icon visible"
          data-testid="clear-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`Search Search suggestions Should show no suggestions when no results 1`] = `
<div>
  <section
    class="search_container home"
  >
    <h1>
      Movie Overflow
    </h1>
    <div
      class="search"
    >
      <div
        class="filters"
      >
        <div
          class="radio_button"
        >
          <input
            checked=""
            data-testid="All-radio"
            id="all"
            name="All"
            type="radio"
            value="all"
          />
          <label
            for="all"
          >
            All
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Movie-radio"
            id="movie"
            name="Movie"
            type="radio"
            value="movie"
          />
          <label
            for="movie"
          >
            Movie
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Tv Series-radio"
            id="tv"
            name="Tv Series"
            type="radio"
            value="tv"
          />
          <label
            for="tv"
          >
            Tv Series
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="People-radio"
            id="person"
            name="People"
            type="radio"
            value="person"
          />
          <label
            for="person"
          >
            People
          </label>
        </div>
      </div>
      <div
        class="input_wrapper"
      >
        <svg
          class="search_icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          type="submit"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
            fill-rule="evenodd"
          />
        </svg>
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search..."
          type="text"
          value="my movie"
        />
        <svg
          class="close_icon visible"
          data-testid="clear-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`Search Search suggestions Should trigger search for suggestions 1`] = `
<div>
  <section
    class="search_container home"
  >
    <h1>
      Movie Overflow
    </h1>
    <div
      class="search"
    >
      <div
        class="filters"
      >
        <div
          class="radio_button"
        >
          <input
            checked=""
            data-testid="All-radio"
            id="all"
            name="All"
            type="radio"
            value="all"
          />
          <label
            for="all"
          >
            All
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Movie-radio"
            id="movie"
            name="Movie"
            type="radio"
            value="movie"
          />
          <label
            for="movie"
          >
            Movie
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Tv Series-radio"
            id="tv"
            name="Tv Series"
            type="radio"
            value="tv"
          />
          <label
            for="tv"
          >
            Tv Series
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="People-radio"
            id="person"
            name="People"
            type="radio"
            value="person"
          />
          <label
            for="person"
          >
            People
          </label>
        </div>
      </div>
      <div
        class="input_wrapper"
      >
        <svg
          class="search_icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          type="submit"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
            fill-rule="evenodd"
          />
        </svg>
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search..."
          type="text"
          value="my movie"
        />
        <svg
          class="close_icon visible"
          data-testid="clear-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
        <div
          class="suggestions false"
        >
          <div
            class="item"
          >
            <h3>
              mockActor
            </h3>
            <div
              class="type"
            >
              person
            </div>
          </div>
          <div
            class="item"
          >
            <h3>
              mockTvSeries
            </h3>
            <div
              class="type"
            >
              tv
            </div>
          </div>
          <div
            class="item"
          >
            <h3>
              mockMovie
            </h3>
            <div
              class="type"
            >
              all
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Search When rendering Should render consistently 1`] = `
<div>
  <section
    class="search_container home"
  >
    <h1>
      Movie Overflow
    </h1>
    <div
      class="search"
    >
      <div
        class="filters"
      >
        <div
          class="radio_button"
        >
          <input
            checked=""
            data-testid="All-radio"
            id="all"
            name="All"
            type="radio"
            value="all"
          />
          <label
            for="all"
          >
            All
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Movie-radio"
            id="movie"
            name="Movie"
            type="radio"
            value="movie"
          />
          <label
            for="movie"
          >
            Movie
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="Tv Series-radio"
            id="tv"
            name="Tv Series"
            type="radio"
            value="tv"
          />
          <label
            for="tv"
          >
            Tv Series
          </label>
        </div>
        <div
          class="radio_button"
        >
          <input
            data-testid="People-radio"
            id="person"
            name="People"
            type="radio"
            value="person"
          />
          <label
            for="person"
          >
            People
          </label>
        </div>
      </div>
      <div
        class="input_wrapper"
      >
        <svg
          class="search_icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          type="submit"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
            fill-rule="evenodd"
          />
        </svg>
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search..."
          type="text"
          value=""
        />
        <svg
          class="close_icon false"
          data-testid="clear-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;
